fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeTwoAggRule.java

```diff
@@ -0,0 +1,133 @@
+//
+// Before:
+//      Aggregate
+//          |
+//       Project
+//          |
+//      Aggregate
+//          |
+//         Node
+//
+// After:
+//      Aggregate
+//          |
+//         Node
+public class MergeTwoAggRule extends TransformationRule {
+    private static final List<String> ALLOW_MERGE_AGGREGATE_FUNCTIONS = Lists.newArrayList(FunctionSet.SUM,
+            FunctionSet.MAX, FunctionSet.MIN, FunctionSet.BITMAP_UNION, FunctionSet.HLL_UNION,
+            FunctionSet.PERCENTILE_UNION);
+
+    public MergeTwoAggRule() {
+        super(RuleType.TF_MERGE_TWO_AGG_RULE, Pattern.create(OperatorType.LOGICAL_AGGR).addChildren(
+                Pattern.create(OperatorType.LOGICAL_PROJECT)
+                        .addChildren(Pattern.create(OperatorType.LOGICAL_AGGR, OperatorType.PATTERN_LEAF))));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggregateAbove = (LogicalAggregationOperator) input.getOp();
+        LogicalProjectOperator project = (LogicalProjectOperator) input.getInputs().get(0).getOp();
+        LogicalAggregationOperator aggregateBelow =
+                (LogicalAggregationOperator) input.getInputs().get(0).getInputs().get(0).getOp();
+
+        // Only handle column prune, forbidden expression
+        if (!project.getColumnRefMap().entrySet().stream().allMatch(e -> e.getKey().equals(e.getValue()))) {
+            return false;
+        }
+
+        if (aggregateBelow.getPredicate() != null || aggregateBelow.hasLimit()) {
+            return false;
+        }
+
+        // The data distribution of the aggregate nodes keeps same
+        if (!aggregateBelow.getGroupingKeys().containsAll(aggregateAbove.getGroupingKeys())) {
+            return false;
+        }
+
+        Map<ColumnRefOperator, CallOperator> aggCallMapAbove = aggregateAbove.getAggregations();
+        Map<ColumnRefOperator, CallOperator> aggCallMapBelow = aggregateBelow.getAggregations();
+
+        if (!aggCallMapAbove.values().stream().allMatch(
+                c -> c.isAggregate() && c.getUsedColumns().cardinality() == 1 && c.getChild(0).isColumnRef())) {
+            return false;
+        }
+
+        for (CallOperator call : aggCallMapAbove.values()) {
+            ColumnRefOperator ref = (ColumnRefOperator) call.getChild(0);
+            if (!ALLOW_MERGE_AGGREGATE_FUNCTIONS.contains(call.getFnName().toLowerCase())) {
+                return false;
+            }
+
+            // max/min on grouping column is equals with direct aggregate on column
+            if (!aggCallMapBelow.containsKey(ref) && !FunctionSet.MAX.equalsIgnoreCase(call.getFnName()) &&
+                    !FunctionSet.MIN.equalsIgnoreCase(call.getFnName())) {
+                return false;
+            }
+
+            if (!aggCallMapBelow.get(ref).getFnName().equalsIgnoreCase(call.getFnName())) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
Whether to check the distinct attributes of functions, for example, 
`select sum(ID_int) from (select sum(distinct ID_int) ID_int from test_basic) a;`

Reply:
Yeah，you are right.